account_level_custom_nameservers.go

```diff
@@ -0,0 +1,162 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type CustomNameserverRecord struct {
+	Type  string `json:"type"`
+	Value string `json:"value"`
+}
+
+type CustomNameserver struct {
+	NSName string `json:"ns_name"`
+	NSSet  int    `json:"ns_set"`
+}
+
+type CustomNameserverResult struct {
+	DNSRecords []CustomNameserverRecord `json:"dns_records"`
+	NSName     string                   `json:"ns_name"`
+	NSSet      int                      `json:"ns_set"`
+	Status     string                   `json:"status"`
+	ZoneTag    string                   `json:"zone_tag"`
+}
+
+type CustomNameserverZoneMetadata struct {
+	Type    string `json:"type"`
+	NSSet   string `json:"ns_set"`
+	Enabled bool   `json:"enabled"`
+}
+
+type customNameserverListResponse struct {
+	Response
+	Result []CustomNameserverResult `json:"result"`
+}
+
+type customNameserverCreateResponse struct {
+	Response
+	Result CustomNameserverResult `json:"result"`
+}
+
+type customNameserverZoneMetadata struct {
+	Response
+	Result CustomNameserverZoneMetadata
+}
+
+// GetAccountCustomNameservers lists an account's custom nameservers.
+//
+// API documentation: https://developers.cloudflare.com/api/operations/account-level-custom-nameservers-list-account-custom-nameservers
+func (api *API) GetAccountCustomNameservers(ctx context.Context, rc *ResourceContainer) ([]CustomNameserverResult, error) {
+	uri := fmt.Sprintf("/accounts/%s/custom_ns", rc.Identifier)
```

Comment:
i'm _thinking_ we can drop the `Account` portion of these method names and instead, rely on the `ResourceContainer` to determine the routes/scope in use.

```suggestion
func (api *API) GetCustomNameservers(ctx context.Context, rc *ResourceContainer) ([]CustomNameserverResult, error) {
    if rc.Level != AccountRouteLevel {
		return []CustomNameserverResult{}, ErrRequiredAccountLevelResourceContainer
	}
    
	uri := fmt.Sprintf("/%s/%s/custom_ns", rc.Level, rc.Identifier)
```

let me confirm with some folks internally if we have any strong objections here.

Reply:
pushed up a commit making this move to better align with the newer conventions.